nimble/host/src/ble_gattc.c

```diff
@@ -3172,17 +3176,17 @@ ble_gattc_read_long_rx_read_rsp(struct ble_gattc_proc *proc, int status,
 
     data_len = OS_MBUF_PKTLEN(*om);
 
-    attr.handle = proc->read_long.handle;
-    attr.offset = proc->read_long.offset;
-    attr.om = *om;
-
-    /* Report partial payload to application. */
-    rc = ble_gattc_read_long_cb(proc, status, 0, &attr);
+    /* Queue Partial Reading. */
```

Comment:
Since this requires extra memory for caching I'd prefer if this would be configurable via syscfg

Reply:
I'd just create new API for this, e.g. `ble_gattc_read_long_full` (or some smarter name) which can be a simple wrapper on current API

other than this patch lgtm

Reply:
So you suggest to add a syscfg to enable/disable full reading and if we enabled full reading we have ble_gattc_read_long_full instead of ble_gattc_read_long. Is this correct?

Reply:
no, my proposal is not to make syscfg but keep old API as-is and add new one with new behavior

Reply:
@andrzej-kaczmarek I have added the new API that perform the Full reading and update the PR accordingly.